block/manager.go

```diff
@@ -605,10 +603,10 @@ func (m *Manager) publishBlock(ctx context.Context) error {
 
 	newState.DAHeight = atomic.LoadUint64(&m.daHeight)
 	// After this call m.lastState is the NEW state returned from ApplyBlock
-	m.lastState = newState
+	m.updateLastState(newState)
 
 	// UpdateState commits the DB tx
-	err = m.store.UpdateState(m.lastState)
+	err = m.store.UpdateState(newState)
```

Comment:
So my last question is how we should be handling this error case. If we are unable to update the state in the store, should we revert the state?

Put another way, should we move this into the `updateLastState` method?

```go
func (m *Manager) updateLastState(s types.State) error {
	m.lastStateMtx.Lock()
	defer m.lastStateMtx.Unlock()
	err = m.store.UpdateState(newState)
        if err != nil {
                return err
        }
	m.lastState = s
        return nil
}
```

This way, we are either fully updating the state or not. In the current implementation I think there is room for a state mismatch in the event of an error from `m.store.UpdateState`

Reply:
That's a great point! Updated the code to reflect these changes. Also thought it made more sense to just call it `updateState` within `Block Manager` :)